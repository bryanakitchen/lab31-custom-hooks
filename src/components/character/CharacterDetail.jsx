import React from 'react';
import PropTypes from 'prop-types';

function CharacterDetail({ character }) {
  return (
    <figure>
      <img src={character.image} alt={character.name} />
      <figcaption>{character.name}</figcaption>     
    </figure>
  );
}
// add more after all is connected
CharacterDetail.propTypes = {
  character: PropTypes.shape({
    name: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired
  })
};

export default CharacterDetail;

