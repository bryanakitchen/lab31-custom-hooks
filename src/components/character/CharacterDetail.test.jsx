/* eslint-disable max-len */
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CharacterDetail from './CharacterDetail';

describe('CharacterDetail component', () => {
  afterEach(() => cleanup());
  it('renders CharacterDetail', () => {
    const { asFragment } = render(
      <CharacterDetail character={{
        name: 'Admiral', 
        image:'https://vignette.wikia.nocookie.net/animalcrossing/images/e/ed/Admiral_NH.png/revision/latest?cb=20200802081138'
      }}
      />);
    expect(asFragment()).toMatchSnapshot();
  });
});
