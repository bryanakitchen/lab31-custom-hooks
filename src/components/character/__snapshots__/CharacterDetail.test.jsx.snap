// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterDetail component renders CharacterDetail 1`] = `
<DocumentFragment>
  <figure
    data-testid="character"
  >
    <img
      alt="Admiral"
      src="https://vignette.wikia.nocookie.net/animalcrossing/images/e/ed/Admiral_NH.png/revision/latest?cb=20200802081138"
    />
    <figcaption>
      Admiral
    </figcaption>
  </figure>
</DocumentFragment>
`;
